code/modules/ninja/suit/suit_initialisation.dm

```diff
@@ -1,102 +1,76 @@
+	sleep(delay)
+	to_chat(U, "<span class='notice'>Extending neural-net interface...\nNow monitoring brain wave pattern...</span>")
+	sleep(delay)
+	if(U.stat==2||U.health<=0)
+		to_chat(U, "<span class='danger'><B>FÄAL ï¿½Rrï¿½R</B>: 344--93#ï¿½&&21 BRï¿½ï¿½N |/|/aVï¿½ PATT$RN <B>RED</B>\nA-A-aBï¿½rTï¿½NG...</span>")
+		unlock_suit()
+		return FALSE
+	lockIcons(U)//Check for icons.
+	U.regenerate_icons()
+	to_chat(U, "<span class='notice'>Linking neural-net interface...\nPattern</span>\green <B>GREEN</B><span class='notice'>, continuing operation.</span>")
+	sleep(delay)
+	to_chat(U, "<span class='notice'>VOID-shift device status: <B>ONLINE</B>.\nCLOAK-tech device status: <B>ONLINE</B>.</span>")
+	sleep(delay)
+	to_chat(U, "<span class='notice'>Primary system status: <B>ONLINE</B>.\nBackup system status: <B>ONLINE</B>.\nCurrent energy capacity: <B>[cell.charge]</B>.</span>")
+	sleep(delay)
+	to_chat(U, "<span class='notice'>All systems operational. Welcome to <B>SpiderOS</B>, [U.real_name].</span>")
+	grant_ninja_verbs()
+	grant_equip_verbs()
+	ntick()
+	s_busy = FALSE
+	return TRUE
 
 
 
 /obj/item/clothing/suit/space/space_ninja/proc/deinitialize(delay = s_delay)
-	if(affecting==loc&&!s_busy)
+	if(affecting==loc)
 		var/mob/living/carbon/human/U = affecting
-		if(!s_initialized)
-			to_chat(U, "<span class='danger'>The suit is not initialized.</span> Please report this bug.")
-			return
 		if(alert("Are you certain you wish to remove the suit? This will take time and remove all abilities.",,"Yes","No")=="No")
-			return
-		if(s_busy)
-			to_chat(U, "<span class='userdanger'>ERROR</span>: You cannot use this function at this time.")
-			return
-		s_busy = 1
-		for(var/i = 0,i<7,i++)
-			switch(i)
-				if(0)
-					to_chat(U, "<span class='notice'>Now de-initializing...</span>")
-					spideros = 0//Spideros resets.
-				if(1)
-					to_chat(U, "<span class='notice'>Logging off, [U:real_name]. Shutting down <B>SpiderOS</B>.</span>")
-					remove_ninja_verbs()
-				if(2)
-					to_chat(U, "<span class='notice'>Primary system status: <B>OFFLINE</B>.\nBackup system status: <B>OFFLINE</B>.</span>")
-				if(3)
-					to_chat(U, "<span class='notice'>VOID-shift device status: <B>OFFLINE</B>.\nCLOAK-tech device status: <B>OFFLINE</B>.</span>")
-					cancel_stealth()//Shutdowns stealth.
-				if(4)
-					to_chat(U, "<span class='notice'>Disconnecting neural-net interface...</span>\green<B>Success</B><span class='notice'>.</span>")
-				if(5)
-					to_chat(U, "<span class='notice'>Disengaging neural-net interface...</span>\green<B>Success</B><span class='notice'>.</span>")
-				if(6)
-					to_chat(U, "<span class='notice'>Unsecuring external locking mechanism...\nNeural-net abolished.\nOperation status: <B>FINISHED</B>.</span>")
-					remove_equip_verbs()
-					unlock_suit()
-					U.regenerate_icons()
-			sleep(delay)
-		s_busy = 0
-	return
\ No newline at end of file
+			return FALSE
+		s_busy = TRUE
+		sleep(delay)
```

Comment:
addtimer is equivalent to spawn, not sleep.

I mean, I suppose I could do this with addtimer, but wouldn't that induce more overhead than a sleep?